src/System.Private.ServiceModel/tests/Common/Infrastructure/src/BridgeClient.cs

```diff
@@ -222,30 +260,48 @@ private static string MakeResourcePutRequest(string resourceName)
             }
         }
 
-        private static string MakeResourceGetRequest(string resourceName)
+        internal static Dictionary<string, string> MakeResourceGetRequest(string resourceName, Dictionary<string, string> requestParameters)
         {
             EnsureBridgeIsRunning();
 
+            string requestParametersResourceName;
+            if (requestParameters != null
+                && requestParameters.TryGetValue(EndpointResourceRequestNameKeyName, out requestParametersResourceName)
+                && !string.Equals(resourceName, requestParametersResourceName))
+            {
+                throw new ArgumentException(
+                    string.Format("A GET request to the bridge included parameter 'name' = '{0}' in its requestParameters, but the 'resourceName' = '{1}' specified was different", requestParametersResourceName, resourceName),
+                    "requestParameters");
+            }
+
             using (HttpClient httpClient = new HttpClient())
             {
+                StringBuilder uriBuilder = new StringBuilder(BridgeResourceEndpointAddress + resourceName);
+                if (requestParameters != null)
+                {
```

Comment:
Nit: consider a helper to compose query string parameters from name/value pairs.  Not critical.